src/org/traccar/WebDataHandler.java

```diff
@@ -95,10 +104,10 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
                 .replace("{valid}", String.valueOf(position.getValid()))
                 .replace("{latitude}", String.valueOf(position.getLatitude()))
                 .replace("{longitude}", String.valueOf(position.getLongitude()))
-                .replace("{altitude}", String.valueOf(position.getAltitude()))
-                .replace("{speed}", String.valueOf(position.getSpeed()))
-                .replace("{course}", String.valueOf(position.getCourse()))
-                .replace("{accuracy}", String.valueOf(position.getAccuracy()))
+                .replace("{altitude}", (position.getAltitude() == null) ? "0" : String.valueOf(position.getAltitude()))
```

Comment:
Are this substitutions to zero obligatory for protocol? Can it be empty string? Just do not know.

Reply:
@tananaev suggested zeroes, I'm fine with both options. Will let one of you have the final word.